scenes/test_object.txt

```diff
@@ -5,17 +5,13 @@
 
 *gosub test_start "get unmodified data"
 *gosub_scene cslib_object get "testobj" 1
-*gosub test_assert (testobj_testfield1 = "Calvin") "=> Expected - Calvin[n/]=> Received - ${testobj_testfield1}[n/]"
+*gosub test_assert (testobj_testfield1 = testobj_1_testfield1) "=> Expected - ${testobj_1_testfield1}[n/]=> Received - ${testobj_testfield1}[n/]"
 *gosub test_finish
 
-*gosub test_start "set (modify) data"
+*gosub test_start "set (modify) and get (modified) data"
 *gosub_scene cslib_object set "testobj" 2 "testfield1" "Locke"
-*gosub test_assert (testobj_2_testfield1 = "Locke") "=> Expected - Locke[n/]=> Received - ${testobj_2_testfield1}[n/]"
-*gosub test_finish
-
-*gosub test_start "get modified data"
 *gosub_scene cslib_object get "testobj" 2
-*gosub test_assert (testobj_testfield1 = "Locke") "=> Expected - Locke[n/]=> Received - ${testobj_testfield1}[n/]"
+*gosub test_assert ((testobj_2_testfield1 = "Locke") and (testobj_testfield1 = testobj_2_testfield1)) "=> Expected - Locke[n/]=> Received - ${testobj_testfield1}[n/]"
```

Comment:
Can you make this two separate test_asserts so on a failure the error message is clear?

Reply:
You suggested yesterday:
_A test can have multiple test_assert statements, so I suggest just moving this one into the test above (and renaming the test to set and get (modified) data accordingly._
Possibly I misunderstood?

Reply:
Sorry, I meant that you can have two _calls_ to 'test_assert', like so:

```
*gosub test_assert (testobj_2_testfield1 = "Locke") "=> Expected - Locke[n/]=> Received - ${testobj_testfield1}[n/]"
*gosub test_assert (testobj_testfield1 = testobj_2_testfield1) "=> Expected - Locke[n/]=> Received - ${testobj_testfield1}[n/]"
```

Functionally it's the same, but if one condition fails, it's easier to catch which one.

This way your output would be:

```
TEST BLA BLA - PASS
=> Expected - Locke
=> Received - Locke
=> Expected - Locke
=> Received - Locke
```

Or maybe:

```
TEST BLA BLA - FAIL
=> Expected - Locke
=> Received - Locke

=> Expected - David
=> Received - Locke
```

Which makes it apparent that it's the second assert that failed, not the first. If both checks are in the same assert, then it's more difficult to see which condition failed.

Reply:
All of that said, I'm not sure I follow what you're testing here anymore. I think it will be easier to follow for both of us once you've renamed the field attributes.